File: 프로그래머스/3/43238. 입국심사/입국심사.py
def solution(n, times):
    low_time = 0
    high_time = max(times) * n
    
    while low_time <= high_time:
        mid_time = (low_time + high_time) // 2
        
        mid_n = 0
        for time in times:
            mid_n += mid_time // time
        
        if mid_n >= n:
            high_time = mid_time - 1
        else:
            low_time = mid_time + 1
    
    return low_time

File: 프로그래머스/3/43238. 입국심사/입국심사.js
function solution(n, times) {
    let low_time = 0;
    let high_time = Math.max(...times) * n;
    
    while (low_time <= high_time) {
        const mid_time = Math.floor((low_time + high_time) / 2);
        
        const mid_n = times.reduce((acc, time) => acc += Math.floor(mid_time / time), 0);

        if (mid_n >= n) high_time = mid_time - 1;
        else low_time = mid_time + 1;
    }
    
    return low_time;
}